check/src/main/scala/rsc/checkbytecode/Main.scala

```diff
@@ -2,12 +2,39 @@
 // Licensed under the Apache License, Version 2.0 (see LICENSE.md).
 package rsc.checkbytecode
 
+import cats.implicits._
+import com.monovore.decline.{CommandApp, Opts}
+import com.monovore.decline.Opts.{argument, flag, option}
 import java.nio.file._
 import rsc.checkbase._
 
-object Main extends SimpleBase[Settings, Path, Path] {
-  def settings(args: List[String]) = {
-    Settings.parse(args)
+object Main extends CommandApp(MainCommand.command)
+
+object MainCommand extends SimpleBase[Settings, Path, Path] {
+  val name = "checkbytecode"
+  val header = "checkbytecode"
+  val opts: Opts[Settings] = {
+    val insArgs = argument[SourceFiles]("files")
+
+    val cpOpt =
+      option[RscClassPath](
+        "classpath",
+        short = "c",
+        help = "classpaths separated by java.io.File.pathSeparator")
+
+    val depsOpt =
+      option[RscClassPath](
+        "dependency classpaths",
+        short = "d",
+        help = "classpaths separated by java.io.File.pathSeparator")
+
+    val quietOpt =
+      flag("quiet", short = "q", help = "Don't output diffs to standard out").orFalse
+
+    (insArgs, cpOpt, depsOpt, quietOpt).mapN { (ins, cp, deps, quiet) =>
+      //TODO, what does the string input for deps look like?
```

Comment:
Seems to be like `checkbytecode --deps X.scala Y.scala -- Z.scala`, compiling `Z.scala` against a classpath containing `X.scala` and `Y.scala`. However, I think that `X.scala:Y.scala` makes more sense.

Reply:
That seems like it should just be a List[Path] then, but I had to artificially transform it into a List[List[Path]] and wasn't sure why.